### DISCUSSION TITLE
Display more information on mouse hover on agent in the web app

### DISCUSSION THREAD

**Discussion by Timocaillou** [mesa/mesa#2080] [General]
Hi! I'm currently making a SIR model with some virus mutation. In the web app, I would like to visualize virus parameters on hovering an agent but don't know how I can do that? 
Thanks in advance, any help is welcome.

**Comment by FoFFolo** [comment]
Hi, if you're using the mesa visualization server to run the simulation, if you want to see information on hover, you have to set the "text" element on the agent_portrayal function(e.g):
```
def agent_portrayal(agent: MyAgent):
    portrayal = {
        "Shape": "circle",
        "Filled": "true",
        "r": 0.5,
        "text": agent.unique_id
    }

    return portrayal
```
But take care that the mesa visualization server is no more actively developed.
If you're using default solara, the hover option is not possible because the chart that you see is made using matplotlib, which render an image, so it's no interactable.
The only way now is to render the chart using Altair chart, which is currently an experimental project for mesa(check the altair.py script in mesa/mesa/experimental/components/altair.py).
I'll give you an example:
```
def agent_portrayal(MyAgent):
    size = 10
    color = "tab:red"
    return {"size": size, "color": color, "Date": datetime.now(), "Unique_id": MyAgent.unique_id}
```
The chart will show a tooltip of every property except for "size", "color", "x" and "y".

```
page = JupyterViz(
    model_class=MyModel,
    model_params=model_params,
    measures=[make_histogram],
    name="My Model",
    agent_portrayal=agent_portrayal,
    space_drawer="altair"
)
```
If you set the parameter "space_drawer='altair'", the chart will be rendered using altair.py.

**Reply by EwoutH** [reply]
Thanks for this extensive reply! @Timocaillou, did this answer you question?
